### DISCUSSION TITLE
podcast syncronization

### DISCUSSION THREAD

**Discussion by mcmikemn** [EnigmaCurry/d.rymcg.tech#50] [Ideas]
I'd like to syncronize podcasts (including play state) between devices, partially as a backup (so I'll have all my subscriptions and know which episodes were litened to, in case I lose one device) and partially to be able to continue listening from different devices. gpodder seemed like the only solution so I create the gpodder branch, but it's a desktop-app access via remote gui, its interface sucks, and it doesn't work well. I don't think we should bother incorporating the gpodder branch into d.rymcg.

**Comment by mcmikemn** [comment]
Why was it so hard to find [audiobookshelf](https://github.com/advplyr/audiobookshelf)? It's a huge step in the right direction: gorgeous interface and mostly intuitive. As far as I can tell it's mostly a podcast (and ebook) proxy: you set up your subscriptions in audiobookshelf (and you can set up automatic new-episode downloads) and then point your podcast app to feeds that audiobookshelf sets up.

There's no play-state syncronization, though - so if I've played half or a full episode or marked an episode as "played" in Antennapod, that info does not get synced back to audiobookshelf. Audiobookshelf does have it's own player, so you can listen to your podcasts from anywhere via the web interface, so that's...something. And they say audiobookshelf can "[s]tream all audio formats on the fly." If it does stream audio, then a player on a phone would actually be playing the media on the server, so the server would be able to know/remember the play state. But Antennapod doesn't seem to have any functionality to stream a feed (it just downloads episodes).

I upvoted a feature request at Antennapod to integrate syncronization with audiobookshelf, but I get the impression it's not going to happen any time soon.

I added an audiobookshelf branch.

**Comment by mcmikemn** [comment]
FYI, searching and playing podcasts via audiobookshelf's web interface on a phone is quite easy. Not as easy as an app, but pretty close. Maybe I don't even need Antennapod?

**Comment by EnigmaCurry** [comment]
Thats awesome!

> FYI, searching and playing podcasts via audiobookshelf's web interface on a phone is quite easy. Not as easy as an app, but pretty close. Maybe I don't even need Antennapod?

I often go on walks in places that don't have any wireless coverage, so streaming doesn't always work.

**Reply by mcmikemn** [reply]
> I often go on walks in places that don't have any wireless coverage, so streaming doesn't always work.

I agree (and on planes). I mentioned that in my feature request with Audiobookshelf.

**Comment by mcmikemn** [comment]
The developer pointed out that an APK is included with every release on github, so the fact that it's not in F-droid doesn't matter. I installed it and it works fine without Google Play services. _And_ it allows you to either stream from your server _or_ download episodes (and he says that when your phone regains connectivity, the app will sync play state of downloaded episodes to the server).
